### PR TITLE
Feature ETP-846: First model of the process modal

### PR DESCRIPTION
**Pull request by Sankku**

### CODE REVIEW

File: packages/MainUI/components/Toolbar/Toolbar.tsx [hunk 1]
```diff
@@ -27,9 +33,36 @@ export const Toolbar: React.FC<ToolbarProps> = ({ windowId, tabId }) => {
...
+              msgText: error instanceof Error ? error.message : 'Unknown error',
+            },
+          },
+        ],
+      });
+    } finally {
+      setIsExecuting(false);
+    }
+  };
+
```
Comment (LuuchoRocha) [inline packages/MainUI/components/Toolbar/Toolbar.tsx:65]: please wrap this in useCallback 😅

File: packages/MainUI/components/Toolbar/Toolbar.tsx [hunk 2]
```diff
@@ -97,7 +135,28 @@ export const Toolbar: React.FC<ToolbarProps> = ({ windowId, tabId }) => {
...
+    <>
+      <TopToolbar {...createToolbarConfig()} />
+      {selectedProcessButton && (
+        <ProcessModal
+          open={openModal}
+          onClose={() => {
+            setOpenModal(false);
+            setSelectedProcessButton(null);
+            setProcessResponse(null);
+          }}
```
Comment (LuuchoRocha) [inline packages/MainUI/components/Toolbar/Toolbar.tsx:148]:
```tsx
const handleClose = useCallback(() => {
  setOpenModal(false);
  setSelectedProcessButton(null);
  setProcessResponse(null);
}, []);

.
.
.

<ProcessModal onClose={handleClose} ... />
```

**Review by LuuchoRocha — CHANGES_REQUESTED** [review]
good work! just two minor things to improve

File: packages/ComponentLibrary/src/components/ProcessModal/styles.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            marginBottom: '0.5rem',
+          },
+          successMessage: {
+            fontSize: '0.875rem',
+            fontWeight: 500,
+            color: theme.palette.baselineColor.neutral[80],
+            marginBottom: '0.5rem',
+          },
+        },
+      } satisfies Record<string, SxProps<Theme>>),
```
Comment (LuuchoRocha) [inline packages/ComponentLibrary/src/components/ProcessModal/styles.ts:88]: satisfies 🤯

**Review by LuuchoRocha — APPROVED** [review]

**Review by NazaTarrili-Futit — APPROVED** [review]

**Pull request merged**